mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1229,14 +1232,17 @@ impl TakerSwap {
         &self,
         taker_payment_hex: Vec<u8>,
         taker_spends_maker_payment_preimage: Vec<u8>,
+        taker_refunds_payment: Vec<u8>,
     ) -> TakerSwapWatcherData {
         TakerSwapWatcherData {
             uuid: self.uuid,
             secret_hash: self.r().secret_hash.into(),
             taker_spends_maker_payment_preimage,
+            taker_refunds_payment,
             swap_started_at: self.r().data.started_at,
             lock_duration: self.r().data.lock_duration,
             taker_coin: self.r().data.taker_coin.clone(),
+            taker_fee_hash: self.r().taker_fee.as_ref().unwrap().tx_hash.0.clone(),
```

Comment:
Is it possible to avoid unwrapping the taker fee?

Reply:
The taker_fee field of the TakerSwap is an Option. When the create_watcher_data method is used, we know for sure that the taker_fee field must have been already set. Maybe I can turn it into a Result by using `ok_or` and handle the error. Or I can use `expect` and explain why the field is expected to exist. Or I can just leave it as it is. What do you think?

Actually this is the case for a lot of fields in `TakerSwap `and `MakerSwap` structs and unwraps are used all around both in `taker_swap` and maker_swap files, so we could also open a small refactoring issue for this.

Reply:
Let's leave it as it is, I seem to overthinking on this unwrap